mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -48,15 +48,361 @@ using namespace mlir::miopen;
+                                     lowerLayerBounds[iter]);
+
+          // overflow logic.
+          auto ifOverflowOp = b.create<scf::IfOp>(
+              loc, b.getIntegerType(32), carryOp, /*withElseRegion=*/true);
+          auto ifOverflowThenBuilder = ifOverflowOp.getThenBodyBuilder();
+          auto updated = ifOverflowThenBuilder.create<SubIOp>(
+              loc, carriedResult, lowerLayerBounds[iter]);
+          ifOverflowThenBuilder.create<scf::YieldOp>(loc, updated.getResult());
+          auto ifOverflowElseBuilder = ifOverflowOp.getElseBodyBuilder();
+          updated = ifOverflowElseBuilder.create<SubIOp>(loc, carriedResult,
+                                                         zeroConstantI32Op);
+          ifOverflowElseBuilder.create<scf::YieldOp>(loc, updated.getResult());
+
+          // updatedResult is by default of i32 type.
+          Value updatedResult = ifOverflowOp.results()[0];
+          lowerIndicesCarryChecked[lowerDim] = updatedResult;
+        }
+        assert(lowerIndicesCarryChecked.size() == lowerIndicesModified.size());
+        lowerIndicesModified.clear();
+        for (unsigned iter = 0; iter < q.size(); ++iter) {
+          int64_t lowerDim = q[iter].template cast<IntegerAttr>().getInt();
+          lowerIndicesModified.push_back(lowerIndicesCarryChecked[lowerDim]);
+        }
+        assert(lowerIndicesModified.size() == q.size());
+      }
+
+      // Set lowerIndicesDiffMap and lowerIndicesUpdatedMap.
+      for (unsigned iter = 0; iter < q.size(); ++iter) {
+        int64_t lowerDim = q[iter].template cast<IntegerAttr>().getInt();
+        int64_t lowerDiff = lowerDiffs[iter];
+        lowerIndicesDiffMap[lowerDim] = lowerDiff;
+        lowerIndicesUpdatedMap[lowerDim] = lowerIndicesModified[iter];
+      }
+    }
+  } // for (auto &mapping : layoutAttr)
+
+  // Convert lowerIndicesDiffMap to lowerIndicesDiff.
+  assert(lowerIndicesDiffMap.size() == lowerLayerShape.size());
+  for (unsigned iter = 0; iter < lowerLayerShape.size(); ++iter)
+    lowerIndicesDiff.push_back(lowerIndicesDiffMap[iter]);
+
+  // Convert lowerIndicesUpdatedMap to lowerIndicesUpdated.
+  assert(lowerIndicesUpdatedMap.size() == lowerLayerShape.size());
+  for (unsigned iter = 0; iter < lowerLayerShape.size(); ++iter)
+    lowerIndicesUpdated.push_back(lowerIndicesUpdatedMap[iter]);
+}
+
+//===----------------------------------------------------------------------===//
+// Utility function to progresseively apply index diff maps to compute the
+// coordinate for the next layer.
+//===----------------------------------------------------------------------===//
+inline void populateLayeredIndicesWithIndexDiffMap(
+    OpBuilder &b, Location loc, const ArrayAttr &layeredTransformMetadata,
+    const SmallVector<AffineMap> &layeredTransform,
+    const SmallVector<SmallVector<Value, 8>, 2> &layeredIndices,
+    const SmallVector<int64_t, 8> &topDiff,
+    SmallVector<SmallVector<int64_t, 8>, 2> &layeredDiffs,
```

Comment:
should be ```Value```

Reply:
Good finding. I have fixed that in the very latest version.